### PR TITLE
[WIP] Add Rice Distributions

### PR DESCRIPTION
**Pull request by souravsingh**
A fix for #1638

### CODE REVIEW

File: pymc3/distributions/continuous.py [hunk 1]
```diff
@@ -1347,3 +1347,48 @@ def logp(self, value):
...
+    Parameters
+    ----------
+    nu : float
+        shape parameter.
+    sd : float
+        standard deviation.
+
+    """
+
+    def __init__(self, sd=None, nu = None, *args, **kwargs):
```
Comment (twiecki) [inline pymc3/distributions/continuous.py]: no space

File: pymc3/distributions/continuous.py [hunk 2]
```diff
@@ -1347,3 +1347,48 @@ def logp(self, value):
...
+        super(Rice, self).__init__(*args, **kwargs)
+        self.sd = sd
```
Comment (twiecki) [inline pymc3/distributions/continuous.py]: need as_theano_tensor (see other dists above)

**Comment by souravsingh** [comment]
The PR would also need tests and an example. What could be a good example for the Rice Distribution?

**Comment by AndreasMittermeier** [comment]
I figured out that a mean and variance is needed in the init function, as well as in the function random, it is necessary to correctly assign the parameters for stats.rice.rvs:

```
    def __init__(self, nu = None, sd=None, *args, **kwargs):
        super(Rice, self).__init__(*args, **kwargs)
        self.nu = tt.as_tenosr_variable(nu)
        self.sd = tt.as_tensor_variable(sd)
        self.mean = sd*np.sqrt(np.pi/2)*tt.exp((-nu**2/(2*sd**2))/2)*((1-(-nu**2/(2*sd**2)))*i0(-(-nu**2/(2*sd**2))/2)-(-nu**2/(2*sd**2))*i1(-(-nu**2/(2*sd**2))/2))
        self.variance = 2*sd**2 + nu**2 - (np.pi*sd**2/2)*(tt.exp((-nu**2/(2*sd**2))/2)*((1-(-nu**2/(2*sd**2)))*i0(-(-nu**2/(2*sd**2))/2)-(-nu**2/(2*sd**2))*i1(-(-nu**2/(2*sd**2))/2)))**2

    def random(self, point=None, size=None, repeat=None):
        nu, sd = draw_values([self.nu, self.sd],
                                  point=point)
        return generate_samples(stats.rice.rvs, b=nu, scale=sd, loc=0, dist_shape=self.shape,
                                size=size)
```

plus in the logp function, nu can be bound to positive values:
```
    def logp(self, value):
        nu = self.nu
        sd = self.sd
        return bound(tt.log(value/(sd**2)*tt.exp(-(value**2+nu**2)/(2*sd**2))*i0(value*nu/(sd**2))), nu>0)
```

**Comment by souravsingh** [comment]
@AndreasMittermeier I have updated the PR according to the code gist.

**Comment by AndreasMittermeier** [comment]
Just a little spelling mistake here:
`self.nu = tt.as_tenosr_variable(nu)`
must be
`self.nu = tt.as_tensor_variable(nu)`

**Comment by twiecki** [comment]
@souravsingh This still needs a unittest. You can check the tests we have for other distributions against scipy, which also has the rice distribution: https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.rice.html.

File: pymc3/distributions/continuous.py [hunk 3]
```diff
@@ -1347,3 +1347,49 @@ def logp(self, value):
...
+    def random(self, point=None, size=None, repeat=None):
+        nu, sd = draw_values([self.nu, self.sd],
+                                  point=point)
+        return generate_samples(stats.rice.rvs, b=nu, scale=sd, loc=0,
+                                dist_shape=self.shape, size=size)
+
+    def logp(self, value):
+        nu = self.nu
+        sd = self.sd
+        return bound(tt.log(value/(sd**2)*tt.exp(-(value**2+nu**2) / (2*sd**2))*i0(value*nu/(sd**2))), nu>0)
```
Comment (twiecki) [inline pymc3/distributions/continuous.py]: white-spaces between math expressions (except squaring).

Comment (twiecki) [reply pymc3/distributions/continuous.py]: new-line for nu > 0.

**Comment by twiecki** [comment]
Thanks @souravsingh!

**Pull request merged**